lib/fs.js

```diff
@@ -62,7 +64,79 @@ if (version[0] < 10 || (version[0] === 10 && version[1] <= 11)) {
   };
 }
 
+// Recursively remove directory
+//   path <string> path to a file or directory to be removed
+//   cb <Function> callback
+const rmRecursive = (path, cb) => {
+  fs.stat(path, (err, stats) => {
+    if (err) {
+      cb(err);
+      return;
+    }
+    if (stats.isDirectory()) {
+      fs.readdir(path, (err, files) => {
+        if (err) {
+          cb(err);
+          return;
+        }
+        if (files.length === 0) {
+          fs.rmdir(path, cb);
+          return;
+        }
+        let erroed = false;
+        let counter = files.length;
+        const callback = err => {
+          if (erroed) return;
+          if (err) {
+            erroed = true;
+            cb(err);
+          } else if (--counter === 0) {
+            fs.rmdir(path, cb);
+          }
+        };
+        files.forEach(f => rmRecursive(pathModule.join(path, f), callback));
+      });
+    } else {
+      fs.unlink(path, cb);
+    }
+  });
+};
+
+// TODO(SemenchenkoVitaliy): remove and use `fs.promises` instead when
+// Node.js 8 is dropped
+let fsPromises;
+if (process.version.slice(1).split('.')[0] >= 10) {
+  fsPromises = fs.promises;
+} else {
+  fsPromises = iter(['stat', 'readdir', 'rmdir', 'unlink']).collectWith(
+    {},
+    (obj, name) => {
+      obj[name] = util.promisify(fs[name]);
+    }
+  );
+}
+
+// Recursively remove directory
+//   path <string> path to a file or directory to be removed
+const rmRecursivePromise = async path =>
+  fsPromises.readdir(path).then(
+    async files => {
+      await Promise.all(
+        files.map(f => rmRecursivePromise(pathModule.join(path, f)))
+      );
+      return fsPromises.rmdir(path);
+    },
+    err => {
+      if (err.code === 'ENOTDIR' || err.code === 'ENOENT') {
```

Comment:
Maybe add a check to avoid skipping `ENOENT` on *nix systems?
```suggestion
      if (err.code === 'ENOTDIR' || process.platform === 'win32' && err.code === 'ENOENT') {
```